streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1027,6 +1064,10 @@ int maybeCommitActiveTasksPerUserRequested() {
         }
     }
 
+    /**
+     * @throws TimeoutException        if task.timeout.ms has been exceeded (non-EOS)
```

Comment:
I think we throw `Timeout` if `commitSync` hits a timeout (ie, just re-throw) independently of `task.timeout.ms`